### PR TITLE
O auth refactor

### PR DESCRIPTION
**Pull request by GavriloDrljaca**
GavriloDrljaca

### CODE REVIEW

File: HRTool/src/main/java/app/Application.java [hunk 1]
```diff
@@ -33,7 +35,7 @@
 @SpringBootApplication
 
 public class Application extends SpringBootServletInitializer implements CommandLineRunner {
```
Comment (nantic) [inline HRTool/src/main/java/app/Application.java:37]: Please remove the everything from the Application.class.

This should be the only thing in Application.class.

```
public static void main(String[] args) {
    SpringApplication.run(Application.class);
}

@Override
protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
    return application.sources(Application.class);
}
```

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@RequestMapping("/employeeRole")
+public class EmployeeRoleRestController {
+	private static final Logger log = Logger.getLogger(EmployeeRoleRestController.class);
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java]: Do not suppress warnings. Fix the issue instead.

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 3]
```diff
@@ -1,35 +1,43 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@RequestMapping("/employeeRole")
+public class EmployeeRoleRestController {
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java:19]: This controller should NOT exist.

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 4]
```diff
@@ -1,35 +1,43 @@
...
+	private static final Logger log = Logger.getLogger(EmployeeRoleRestController.class);
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public ResponseEntity getRole(){
+
+		if(SecurityContextHolder.getContext().getAuthentication().getPrincipal() != null){
+			UserDetails ud = (UserDetails)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java:27]: Use full variable names.

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 5]
```diff
@@ -1,35 +1,43 @@
...
+@RequestMapping("/employeeRole")
+public class EmployeeRoleRestController {
+	private static final Logger log = Logger.getLogger(EmployeeRoleRestController.class);
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public ResponseEntity getRole(){
+
+		if(SecurityContextHolder.getContext().getAuthentication().getPrincipal() != null){
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java:26]: This is unnecessary if you annote the method appropriately with [at]PreAuthorize("isAuthenticated()")

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 6]
```diff
@@ -1,35 +1,43 @@
...
+
+
+@RestController
+@RequestMapping("/employeeRole")
+public class EmployeeRoleRestController {
+	private static final Logger log = Logger.getLogger(EmployeeRoleRestController.class);
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java:23]: What is the point of this?

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 7]
```diff
@@ -1,35 +1,43 @@
...
+	public ResponseEntity getRole(){
+
+		if(SecurityContextHolder.getContext().getAuthentication().getPrincipal() != null){
+			UserDetails ud = (UserDetails)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+			ArrayList<SimpleGrantedAuthority> sga = (ArrayList<SimpleGrantedAuthority>) ud.getAuthorities();
+			
+			log.info("SGA : "+sga.get(0).getAuthority());
+			return new ResponseEntity("{\"role\": \""+sga.get(0).getAuthority()+"\"}", HttpStatus.OK);
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java]: Do not serialize manually.

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 8]
```diff
@@ -1,143 +1,143 @@
...
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+@RestController
+@Configuration
+@ComponentScan("security")
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java]: This is not needed. [at]RestController marks this as a component.

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 9]
```diff
@@ -1,143 +1,143 @@
...
+@RequestMapping("/signin")
+public class GoogleLoginVerifierController {
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java:39]: This controller should NOT exist. Methods should be moved to employeeController.

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 10]
```diff
@@ -1,143 +1,143 @@
...
+
+	private static final Logger log = Logger
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java:41]: log to logger

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 11]
```diff
@@ -1,143 +1,143 @@
...
+					Arrays.asList("282087479252-4v31a07nrjnmfganchk4i1btpvoprjro.apps.googleusercontent.com"))
+			.build();
+
+	@Autowired
+	private EmployeeRepository employeeRepository;
+
+	@Autowired
+	UserDetailsServiceImpl userDetailsService;
+
+	@RequestMapping(value = "/login", method = RequestMethod.POST)
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java:58]: Should be changed to googleLogin so its not confused with the /login method already provided by Spring Security.

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 12]
```diff
@@ -1,143 +1,143 @@
...
+		Payload payload = null;
+		UserDetails uDetails;
+
+		try {
+			idToken = verifier.verify(id_Token);
+
+			if (idToken != null) {
+				payload = idToken.getPayload();
+
+				log.info("PAYLOAD: " + payload);
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java]: What is the purpose of logging these fields?

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java [hunk 13]
```diff
@@ -1,35 +1,43 @@
...
+	private static final Logger log = Logger.getLogger(EmployeeRoleRestController.class);
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	@RequestMapping(method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public ResponseEntity getRole(){
+
+		if(SecurityContextHolder.getContext().getAuthentication().getPrincipal() != null){
+			UserDetails ud = (UserDetails)SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+			ArrayList<SimpleGrantedAuthority> sga = (ArrayList<SimpleGrantedAuthority>) ud.getAuthorities();
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java:28]: UserDetails should implement a method that returns roles.

File: HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java [hunk 14]
```diff
@@ -1,143 +1,143 @@
...
+@ComponentScan("security")
+@RequestMapping("/signin")
+public class GoogleLoginVerifierController {
+
+	private static final Logger log = Logger
+			.getLogger(GoogleLoginVerifierController.class);
+
+	NetHttpTransport transport = new NetHttpTransport();
+
+	private GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(
```
Comment (nantic) [inline HRTool/src/main/java/app/controllers/GoogleLoginVerifierController.java:46]: There is a lot simpler way of doing all of this. Please check:
google-api-services-oauth2 and google-api-services-plus

File: HRTool/src/main/java/security/PreAuthenticatedProcessingFilter.java [hunk 15]
```diff
@@ -1,104 +1,86 @@
...
+import app.repository.EmployeeRepository;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+
+public class PreAuthenticatedProcessingFilter extends AbstractPreAuthenticatedProcessingFilter{
```
Comment (nantic) [inline HRTool/src/main/java/security/PreAuthenticatedProcessingFilter.java:25]: This entire part is not needed.

File: HRTool/src/main/java/security/Security.java [hunk 16]
```diff
@@ -59,14 +48,10 @@ public PreUserDetailService preUserDetailService(){
     	return ser;
     }
     
-    
-    
     @Bean
     public PreAuthenticatedProcessingFilter preFilter() throws Exception {
```
Comment (nantic) [inline HRTool/src/main/java/security/Security.java:53]: Also not needed.

File: HRTool/src/main/java/security/Security.java [hunk 17]
```diff
@@ -59,14 +48,10 @@ public PreUserDetailService preUserDetailService(){
     	return ser;
```
Comment (nantic) [inline HRTool/src/main/java/security/Security.java:49]: Can be written more compact.

File: HRTool/src/main/java/app/Application.java (hunk 1 above)
Comment (GavriloDrljaca) [inline HRTool/src/main/java/app/Application.java:37]: Ok, but my current task is Security and OAuth

File: HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java (hunk 2 above)
Comment (GavriloDrljaca) [inline HRTool/src/main/java/app/controllers/EmployeeRoleRestController.java]: Should i Put "ResponseEntitity<?>" or "ResponseEntity" becouse Srdjan Radulovic said that i should remove <>?

**Pull request closed without merging**